src/toast/ops/elevation_noise.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) 2015-2020 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import copy
+
+import numpy as np
+
+import traitlets
+
+from ..utils import Environment, Logger
+
+from ..timing import function_timer, Timer
+
+from ..noise_sim import AnalyticNoise
+
+from ..traits import trait_docs, Int, Unicode, Float, Bool, Instance, Quantity
+
+from .. import qarray as qa
+
+from .operator import Operator
+
+
+@trait_docs
+class ElevationNoise(Operator):
+    """Modify detector noise model based on elevation.
+
+    This adjusts the detector NET in a noise model based on the median elevation of
+    each detector in each observation.
+
+    The new NET value is computed as:
+
+    .. math::
+        NET_{new} =  1.0e-12 x f_{sample} x (a / sin(el) + b)^2 / NET_{old}
+
+    Where the old NET is estimated from the high frequency samples of the PSD.
+
+    NOTE: since this operator generates a new noise model for all detectors, you
+    should specify all detectors you intend to use downstream when calling exec().
+
+    If the view trait is not specified, then this operator will use the same data
+    view as the detector pointing operator when computing the pointing matrix pixels
+    and weights.
+
+    If the output model is not specified, then the input is modified in place.
+
+    """
+
+    # Class traits
+
+    API = Int(0, help="Internal interface version for this operator")
+
+    noise_model = Unicode(
+        "noise_model", help="The observation key containing the input noise model"
+    )
+
+    out_model = Unicode(
+        None, allow_none=True, help="Create a new noise model with this name"
+    )
+
+    detector_pointing = Instance(
+        klass=Operator,
+        allow_none=True,
+        help="Operator that translates boresight Az / El pointing into detector frame",
+    )
+
+    view = Unicode(
+        None, allow_none=True, help="Use this view of the data in all observations"
+    )
+
+    noise_a = Float(
+        None,
+        allow_none=True,
+        help="Evaluate noise PSD as (a / sin(el) + b) ** 2 * fsample * 1e-12",
```

Comment:
This expression is just the NET, not the full PSD. Also, somewhere we should define that `(a / sin(el) + b)` yields the NET in uK_CMB rts.

Reply:
Class docstring is updated.  These traits also now have units.